commit 05bc28a1f991f48856ddb4adc655fac592b2abd0
Author: Bin Liu <b-liu@ti.com>
Date:   Tue Dec 18 07:58:05 2018 -0600

    usb: musb: dsps: fix runtime pm for peripheral mode
    
    [ Upstream commit 54578ee883e34d2d1c518d48f1c1e2dd3f387188 ]
    
    Since the runtime PM support was added in musb, dsps relies on the timer
    calling otg_timer() to activate the usb subsystem. However the driver
    doesn't enable the timer for peripheral port, then the peripheral port is
    unable to be enumerated by a host if the other usb port is disabled or in
    peripheral mode too.
    
    So let's start the timer for peripheral port too.
    
    Fixes: ea2f35c01d5e ("usb: musb: Fix sleeping function called from invalid context for hdrc glue")
    Acked-by: Tony Lindgren <tony@atomide.com>
    Signed-off-by: Bin Liu <b-liu@ti.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
    Signed-off-by: Sasha Levin <sashal@kernel.org>

diff --git a/drivers/usb/musb/musb_dsps.c b/drivers/usb/musb/musb_dsps.c
index 1e6d78b1334e..403eb97915f8 100644
--- a/drivers/usb/musb/musb_dsps.c
+++ b/drivers/usb/musb/musb_dsps.c
@@ -181,9 +181,11 @@ static void dsps_musb_enable(struct musb *musb)
 
 	musb_writel(reg_base, wrp->epintr_set, epmask);
 	musb_writel(reg_base, wrp->coreintr_set, coremask);
-	/* start polling for ID change in dual-role idle mode */
-	if (musb->xceiv->otg->state == OTG_STATE_B_IDLE &&
-			musb->port_mode == MUSB_OTG)
+	/*
+	 * start polling for runtime PM active and idle,
+	 * and for ID change in dual-role idle mode.
+	 */
+	if (musb->xceiv->otg->state == OTG_STATE_B_IDLE)
 		dsps_mod_timer(glue, -1);
 }
 
@@ -254,6 +256,10 @@ static int dsps_check_status(struct musb *musb, void *unused)
 				musb->xceiv->otg->state = OTG_STATE_A_IDLE;
 				MUSB_HST_MODE(musb);
 			}
+
+			if (musb->port_mode == MUSB_PERIPHERAL)
+				skip_session = 1;
+
 			if (!(devctl & MUSB_DEVCTL_SESSION) && !skip_session)
 				musb_writeb(mregs, MUSB_DEVCTL,
 					    MUSB_DEVCTL_SESSION);
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1161_05bc28a1f991f48856ddb4adc655fac592b2abd0.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1161_05bc28a1f991f48856ddb4adc655fac592b2abd0.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
